pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -90,9 +92,11 @@
 
   // Pre-calculate the aggregation functions and columns for the query so that it can be shared across all the segments
   private AggregationFunction[] _aggregationFunctions;
-  private List<Pair<AggregationFunction, FilterContext>> _filteredAggregationFunctions;
+
```

Comment:
Can we keep this list and remove the `FilterableAggregationFunction`? We should avoid adding this special aggregation function if possible because the filter is not part of the aggregation

Reply:
FilterableAggregationFunction provides a nice abstraction for defining the sub aggregation, filter and associated context (for collecting same predicate aggregation functions together) without requiring a lot of code ripping in QueryContext. It also helps maintain the order aggregations as present in the original SELECT list.

FilterableAggregationFunction is limited to the execution of filtered aggregates and is not evident anywhere else. I would advocate keeping this class for simplicity.

Reply:
I see your point, but my concern is that aggregation function and filter are logically two independent concept, and embedding filter into an aggregation function could cause confusion. If we need to associate some extra attributes to an aggregation function, I'd suggest adding a wrapper class instead of implementing a special `AggregationFunction`.
Based on the current implementation, I feel `Pair` itself should be enough (only need to associate the `FilterContext` with the `AggregationFunction`).  To maintain the order of the aggregations, we may add pairs with `null` `FilterContext`

Reply:
Ok, removed FilterableAggregationFunction and replaced with a pair list. However, I am not removing _aggregationFunctions since it is widely used.